ResearchKit/Common/ORKAnswerFormat.m

```diff
@@ -1864,19 +1885,29 @@ - (instancetype)copyWithZone:(NSZone *)zone {
 }
 
 - (BOOL)isAnswerValid:(id)answer {
-    BOOL isValid = NO;
     if ([answer isKindOfClass:[NSString class]]) {
         return [self isAnswerValidWithString:(NSString *)answer];
     }
-    return isValid;
+    return NO;
```

Comment:
Avoid double return. Keep `isValid` boolean and just assign it inside the if statement.